pkg/api/types.go

```diff
@@ -2049,10 +2049,11 @@ type SecurityContext struct {
 	// RunAsUser is the UID to run the entrypoint of the container process.
 	RunAsUser *int64 `json:"runAsUser,omitempty"`
 
-	// RunAsNonRoot indicates that the container should be run as a non-root user.  If the RunAsUser
-	// field is not explicitly set then the kubelet may check the image for a specified user or
-	// perform defaulting to specify a user.
-	RunAsNonRoot bool
+	// RunAsNonRoot indicates that the container should be run as a non-root user.
+	// Optional.  If true, the Kubelet will validate the image at runtime to ensure that it
```

Comment:
Yes, remove optional.

The occurrence that exist I think are mostly cruft from mechanically merging Go comments with description tag strings.

Speaking of which, omit all the field names, as well. We shouldn't refer to any Go field names nor types in the comments. Users shouldn't have to know the Go code.